src/lib.rs

```diff
@@ -106,11 +130,87 @@ impl SimpleMTKCore for SimpleMTK {
         self.token_id = self.token_id.saturating_add(1);
     }
 
+    /// Burns a token.
+    ///
+    /// Requirements:
+    /// * sender MUST have sufficient amount of token.
+    ///
+    /// Arguments:
+    /// * `id`: Token ID.
+    /// * `amount`: Token's amount to be burnt.
     fn burn(&mut self, id: TokenId, amount: u128) {
         MTKCore::burn(self, vec![id], vec![amount]);
         let sup = self.supply(id);
         let mut _balance = self
             .supply
             .insert(self.token_id, sup.saturating_sub(amount));
     }
+
+    /// Transforms FT tokens to multiple NFTs.
+    ///
+    /// Requirements:
+    /// * a sender MUST have sufficient amount of tokens to burn,
+    /// * a sender MUST be the owner.
+    ///
+    /// Arguments:
+    /// * `id`: Token ID.
+    /// * `amount`: Token's amount to be burnt.
+    /// * `accounts`: To which accounts to mint NFT.
+    /// * `nft_ids`: NFTs' IDs to be minted.
+    /// * `nfts_metadata`: NFT's metadata.
+    fn transform(
+        &mut self,
+        id: TokenId,
+        amount: u128,
+        accounts: Vec<ActorId>,
+        mut nft_ids: Vec<Vec<TokenId>>,
+        nfts_metadata: Vec<Vec<Option<TokenMetadata>>>,
+    ) {
+        // prechecks
+        if nft_ids.len() != nfts_metadata.len() {
+            panic!("MTK: number of ids, accounts and metadata pieces MUST be the same.");
+        }
+        let mut nft_count = 0;
+        for (_, ids) in nft_ids.iter().enumerate() {
+            nft_count += ids.len();
+        }
+        if amount as usize != nft_count {
+            panic!("MTK: amount of burnt tokens MUST be equal to the amount of nfts.");
+        }
+
+        // burn FT (not to produce another message - just simply use burn_impl)
+        self.assert_can_burn(&msg::source(), &id, amount);
+        self.burn_impl(&id, amount);
+
+        let sup = self.supply(id);
+        let mut _balance = self
+            .supply
+            .insert(self.token_id, sup.saturating_sub(amount));
+        let mut ids = vec![];
+        // now we need to mint NFTs (same - just use mint_implementation to remove excessive usage)
+        for (idx, metadata) in nfts_metadata.iter().enumerate() {
+            let amount = nft_ids[idx].len();
+            if accounts[idx] == ActorId::zero() {
+                panic!("MTK: Mint to zero address")
+            }
+            if nft_ids[idx].len() != amount {
```

Comment:
I don't understand this part.
```rust
let amount = nft_ids[idx].len();
if nft_ids[idx].len() != amount ...
```

Reply:
Yeah, should be metadata.len() for amount.